From 3adc90f1f16f77b6b8cd6ee39bfc41ab4f9974ab Mon Sep 17 00:00:00 2001
From: Togan Muftuoglu <toganm@opensuse.org>
Date: Sat, 27 Oct 2012 19:13:01 +0200
Subject: [PATCH] string-to-number

String-to-int function is obsolute since Emacs 22.1 Therefore update
the function to newer emacs functions

Signed-off-by: Togan Muftuoglu <toganm@opensuse.org>
---
 rep-debugger.el | 46 +++++++++++++++++++++++-----------------------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/rep-debugger.el b/rep-debugger.el
index db241e2..6bb4706 100644
--- a/rep-debugger.el
+++ b/rep-debugger.el
@@ -21,7 +21,7 @@
 ;; You should have received a copy of the GNU General Public License
 ;; along with Librep; see the file COPYING.  If not, write to the
 ;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
-;; Boston, MA 02110-1301 USA. 
+;; Boston, MA 02110-1301 USA.
 
 ;; This file is the perldb portions of gud.el with trivial substitutions
 ;; to make it work with rep..
@@ -49,20 +49,20 @@
 
     ;; Process all the complete markers in this chunk.
     (while (string-match "\032\032\\(\\([a-zA-Z]:\\)?[^:\n]*\\):\\([0-9]*\\):.*\n"
-			 gud-marker-acc)
+             gud-marker-acc)
       (setq
 
        ;; Extract the frame position from the marker.
        gud-last-frame
        (cons (substring gud-marker-acc (match-beginning 1) (match-end 1))
-	     (string-to-int (substring gud-marker-acc
-				       (match-beginning 3)
-				       (match-end 3))))
+             (string-to-number (substring gud-marker-acc
+                                          (match-beginning 3)
+                                          (match-end 3))))
 
        ;; Append any text before the marker to the output we're going
        ;; to return - we don't include the marker in this text.
        output (concat output
-		      (substring gud-marker-acc 0 (match-beginning 0)))
+              (substring gud-marker-acc 0 (match-beginning 0)))
 
        ;; Set the accumulator to the remaining text.
        gud-marker-acc (substring gud-marker-acc (match-end 0))))
@@ -73,17 +73,17 @@
     ;; know the full marker regexp above failed, it's pretty simple to
     ;; test for marker starts.
     (if (string-match "\032.*\\'" gud-marker-acc)
-	(progn
-	  ;; Everything before the potential marker start can be output.
-	  (setq output (concat output (substring gud-marker-acc
-						 0 (match-beginning 0))))
+    (progn
+      ;; Everything before the potential marker start can be output.
+      (setq output (concat output (substring gud-marker-acc
+                         0 (match-beginning 0))))
 
-	  ;; Everything after, we save, to combine with later input.
-	  (setq gud-marker-acc
-		(substring gud-marker-acc (match-beginning 0))))
+      ;; Everything after, we save, to combine with later input.
+      (setq gud-marker-acc
+        (substring gud-marker-acc (match-beginning 0))))
 
       (setq output (concat output gud-marker-acc)
-	    gud-marker-acc ""))
+        gud-marker-acc ""))
 
     output))
 
@@ -106,17 +106,17 @@ The directory containing FILE becomes the initial working directory
 and source-file directory for your debugger."
   (interactive
    (list (read-from-minibuffer "Run rep debugger (like this): "
-			       (if (consp gud-rep-history)
-				   (car gud-rep-history)
-				 (concat gud-rep-command-name
-					 " "
-					 (buffer-file-name)
-					 " "))
-			       nil nil
-			       '(gud-rep-history . 1))))
+                   (if (consp gud-rep-history)
+                   (car gud-rep-history)
+                 (concat gud-rep-command-name
+                     " "
+                     (buffer-file-name)
+                     " "))
+                   nil nil
+                   '(gud-rep-history . 1))))
 
   (gud-common-init command-line 'gud-rep-massage-args
-		   'gud-rep-marker-filter 'gud-rep-find-file)
+           'gud-rep-marker-filter 'gud-rep-find-file)
 
 ;  (gud-def gud-break  "b %l"         "\C-b" "Set breakpoint at current line.")
 ;  (gud-def gud-remove "d %l"         "\C-d" "Remove breakpoint at current line")
-- 
1.8.0

